tools/reports/messages/whatsapp.js

```diff
@@ -0,0 +1,117 @@
+const fileHash = require('../../util/backup_filehash')
+const appleTimestamp = require('../../util/apple_timestamp')
+const { queryAll } = require('../../util/query')
+const { parallel } = require('../../util/async')
+
+const WHATSAPP_DB = fileHash(
+  'ChatStorage.sqlite',
+  'AppDomainGroup-group.net.whatsapp.WhatsApp.shared'
+)
+
+module.exports = {
+  version: 4,
+  name: 'messages.whatsapp',
+  description: `List WhatsApp messages`,
+  requiresBackup: true,
+
+  async run (lib, { backup }) {
+    const database = await backup.openDatabase(WHATSAPP_DB)
+
+    const whatsappTextMessages = await getWhatsappTextMessages(database)
+
+    return parallel(
+      (row) => includesParticipants(database, row),
+      whatsappTextMessages
+    )
+  },
+
+  output: {
+    id: (row) => row.Z_PK,
+    type: (row) => getType(row),
+    chatID: (row) => row.ZCHATSESSION,
+    content: (row) => row.ZTEXT,
+    sender: (row) => row.ZFROMJID, // If it's null, it means it's the phone's owner
+    receiver: (row) => row.ZTOJID, // If it's null, it means it's the phone's owner
+    timestamp: (row) => appleTimestamp.toUnixTimeStamp(row.ZMESSAGEDATE),
+    interlocutor: (row) => row.ZPARTNERNAME, // The other group or person the device is talking to
```

Comment:
What's the behaviour of this field exactly? I mean, is it a phone number always or is it something else?

Reply:
It’s a JID, a kind of identifier where the phone number is followed by the domain, for example, 3469456567@whatsapp.group.